superset-frontend/src/SqlLab/actions/sqlLab.js

```diff
@@ -869,14 +870,33 @@ export function saveQuery(query) {
           query,
           result: convertQueryToClient(result.json.item),
         });
-        dispatch(addSuccessToast(t('Your query was saved')));
         dispatch(queryEditorSetTitle(query, query.title));
+        return convertQueryToClient(result.json.item);
       })
       .catch(() =>
         dispatch(addDangerToast(t('Your query could not be saved'))),
       );
 }
 
+export function addSavedQueryToTabState(queryEditor, savedQuery) {
+  return function (dispatch) {
```

Comment:
```suggestion
export const addSavedQueryToTabState = (queryEditor, savedQuery) => dispatch => {
```

If you turn this to an arrow function, you can clean this code a little. You'll also need to remove a closing bracket if you do this.